dhall/src/Dhall/Import/Types.hs

```diff
@@ -47,6 +48,18 @@ instance Pretty Chained where
 data ImportSemantics = ImportSemantics
     { importSemantics :: Expr Src X
     -- ^ The fully resolved import, typechecked and alpha-beta-normal.
+    , semanticHash :: Crypto.Hash.Digest Crypto.Hash.SHA256
+    -- ^ Its semantic hash, i.e. the hash of its encoded normal form.
+    }
+
+data ResolvedExpr = ResolvedExpr
+    { resolvedExpr :: Expr Src X
+    -- ^ The fully resolved expression, i.e. any imports where replaced by their
+    --   semantics. Note that the resolved expression has not yet been
+    --   typechecked nor normalised!
+    , imports :: [ImportSemantics]
```

Comment:
Why do you include the `imports` when hashing if they are already included as subexpressions of `resolvedExpr`?

Reply:
Hah, never thought about that! That makes the code simpler as well. Thanks!